cfme/services/myservice.py

```diff
@@ -1,86 +1,158 @@
-        if icon_href:
-            return InfoBlock.icon_href(*properties)
-        else:
-            return InfoBlock.text(*properties)
+
+class ServiceRetirementForm(MyServicesView):
+    title = Text('#explorer_title_text')
+
+    retirement_date = Calendar('retirementDate')
+    retirement_warning = BootstrapSelect('retirement_warn')
+
+
+class ServiceEditForm(MyServicesView):
+    title = Text('#explorer_title_text')
+
+    name = Input(name='name')
+    description = Input(name='description')
+
+
+class SetOwnershipForm(MyServicesView):
+    title = Text('#explorer_title_text')
+
+    select_owner = BootstrapSelect('user_name')
+    select_group = BootstrapSelect('group_name')
+
+
+class EditTagsForm(MyServicesView):
+    title = Text('#explorer_title_text')
+
+    select_tag = BootstrapSelect('tag_cat')
+    select_value = BootstrapSelect('tag_add')
+
+
+class MyServiceDetailView(MyServicesView):
+    title = Text("#explorer_title_text")
+
+    properties_detail = SummaryTable(title='Properties')
+    lifecycle_detail = SummaryTable(title='Lifecycle')
+    relationships_detail = SummaryTable(title='Relationships')
+    vm_detail = SummaryTable(title='Totals for Service VMs ')
+    smart_mgmnt_detail = SummaryTable(title='Smart Management')
+
+    @property
+    def is_displayed(self):
+        return (
+            self.in_explorer and self.myservice.is_opened and
+            self.title.text == 'Service "{}"'.format(self.context['object'].name)
+        )
+
+
+class EditMyServiceView(ServiceEditForm):
+    title = Text("#explorer_title_text")
+
+    save_button = Button('Save')
+    reset_button = Button('Reset')
+    cancel_button = Button('Cancel')
+
+    @property
+    def is_displayed(self):
+        return (
+            self.in_explorer and self.myservice.is_opened and
+            self.title.text == 'Editing Service "{}"'.format(self.context['object'].name)
+        )
+
+
+class SetOwnershipView(SetOwnershipForm):
+    title = Text("#explorer_title_text")
+
+    save_button = Button('Save')
+
+    @property
+    def is_displayed(self):
+        return (
+            self.in_explorer and self.myservice.is_opened and
+            self.title.text == 'Set Ownership of Service "{}"'.format(self.context['object'].name)
+        )
+
+
+class EditTagsView(EditTagsForm):
+    title = Text("#explorer_title_text")
+
+    save_button = Button('Save')
+
+    @property
+    def is_displayed(self):
+        return (
+            self.in_explorer and self.myservice.is_opened and
+            self.title.text == 'Edit Tags of Service "{}"'.format(self.context['object'].name)
+        )
+
+
+class ServiceRetirementView(ServiceRetirementForm):
+    title = Text("#explorer_title_text")
+
+    save_button = Button('Save')
+
+    @property
+    def is_displayed(self):
+        return (
+            self.in_explorer and self.myservice.is_opened and
+            self.title.text == 'Set/Remove retirement date for Service')
```

Comment:
You need to add the check for which service is this view. Probably check what is selected in the accordion's tree.